### PR TITLE
woc

### PR DESCRIPTION
**Pull request by Lumos000**
无选做

### CODE REVIEW

File: src/ViewModel/NextUpViewModel.cpp [hunk 1]
```diff
@@ -191,14 +192,23 @@ Song NextUpViewModel::getNextSong() {
     }
     case PlayMode::ListRepeat: {
         // TODO
+        auto index = model.indexOf(song);
+        index++;
+        song = model[index];
```
Comment (ArcticLampyrid) [inline src/ViewModel/NextUpViewModel.cpp:197]: What if `index == size - 1`?

File: src/ViewModel/NextUpViewModel.cpp [hunk 2]
```diff
@@ -191,14 +192,23 @@ Song NextUpViewModel::getNextSong() {
...
     case PlayMode::RepeatOne: {
         // TODO
+        song.id = playingSong.id;
         break;
     }
     case PlayMode::Shuffle: {
         // TODO
+        auto index = model.indexOf(song);
+        srand(QTime(0,0,0).secsTo(QTime::currentTime()));
+        auto random = rand() % model.count();
```
Comment (ArcticLampyrid) [inline src/ViewModel/NextUpViewModel.cpp:209]: In fact, using the modulus operator cannot obtain the uniform distribution.
Suggest using modern methods provided by C++ or Qt.

File: src/ViewModel/SongLyricViewModel.h [hunk 3]
```diff
@@ -28,6 +28,8 @@ class SongLyricViewModel : public QAbstractListModel {
 
     QHash<int, QByteArray> roleNames() const override;
 
+    QList<SongLyric> ProcessLyric (QString);
```
Comment (ArcticLampyrid) [inline src/ViewModel/SongLyricViewModel.h:31]:
```suggestion
    QList<SongLyric> ProcessLyric(QString);
```

Follow our code style defined in `clang-format.yaml`, if there is no specific reason.

Comment (Serein207) [reply src/ViewModel/SongLyricViewModel.h:31]: use **pass by const reference** for large object if you don't expect to change it :eyes:

**Pull request closed without merging**